### PR TITLE
NimBLE host: Make set Packet Length API public

### PR DESCRIPTION
**Pull request by prasad-alatkar**
This PR makes API setting packet length (OGF = 0x08, OCF = 0x0022) in controller from host public.

### CODE REVIEW

**Comment by prasad-alatkar** [comment]
I am a little surprised and confused with below build error as there is `#include "host/ble_gap.h"` in btshell app.

```
Error: apps/btshell/src/main.c: In function 'btshell_datalen':

apps/btshell/src/main.c:1922:10: error: implicit declaration of function 'ble_hs_hci_util_set_data_len'; did you mean 'ble_hs_hci_util_set_random_addr'? [-Werror=implicit-function-declaration]

     rc = ble_hs_hci_util_set_data_len(conn_handle, tx_octets, tx_time);

          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~

          ble_hs_hci_util_set_random_addr

cc1: all warnings being treated as errors

```

**Comment by prasad-alatkar** [comment]
@sjanc Could you please also take a quick look here ?

File: nimble/host/include/host/ble_gap.h [hunk 1]
```diff
@@ -1214,6 +1214,23 @@ struct ble_gap_ext_adv_params {
...
+ *                         should use for a new connection (Range
+ *                         0x001B-0x00FB).
+ * @param tx_time          The preferred maximum number of microseconds that the local Controller
+ *                         should use to transmit a single link layer packet
+ *                         (Range 0x0148-0x4290).
+ *
+ * @return              0 on success,
+ *                      other error code on failure.
+ */
+int ble_hs_hci_util_set_data_len(uint16_t conn_handle, uint16_t tx_octets,
```
Comment (sjanc) [inline nimble/host/include/host/ble_gap.h:1231]: I've just keep GAP name convention, so this function should be renamed to ble_gap_set_data_len()

Comment (sjanc) [reply nimble/host/include/host/ble_gap.h:1231]: And for build error, this should not be under #if MYNEWT_VAL(BLE_EXT_ADV)

maybe just place it ble_gap_update_params()

Comment (prasad-alatkar) [reply nimble/host/include/host/ble_gap.h]: @sjanc Thank you for valuable inputs :). I have made relevant changes.

**Comment by sjanc** [comment]
looks good,  could you just squash those into single commit? (it makes git bisect a bit easier in future)

**Comment by prasad-alatkar** [comment]
> looks good, could you just squash those into single commit? (it makes git bisect a bit easier in future)

Sure, sure!!

**Review by sjanc — APPROVED** [review]
thanks:-)

**Pull request merged**